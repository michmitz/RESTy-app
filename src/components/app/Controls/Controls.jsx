import React from 'react';
import PropTypes from 'prop-types';

const Controls = ({ url, method, body, onChange, onSubmit }) => (
  <div className="controls-div">

      <form onSubmit={onSubmit} className="method-form">
        <div className="radio">
          <label>
            <input 
              type="radio" 
              name="method" 
              value="GET" 
              onChange={onChange} 
              checked={method === 'GET'} />
            GET
          </label>
        </div>

        <div className="radio">
          <label>
            <input 
              type="radio" 
              name="method" 
              value="POST" 
              onChange={onChange} 
              checked={method === 'POST'} />
            POST
          </label>
        </div>

        <div className="radio">
          <label>
            <input 
              type="radio" 
              name="method" 
              value="PUT" 
              onChange={onChange} 
              checked={method === 'PUT'} />
            PUT
          </label>
        </div>

        <div className="radio">
          <label>
            <input 
              type="radio" 
              name="method" 
              value="DELETE" 
              onChange={onChange} 
              checked={method === 'DELETE'} />
            DELETE
          </label>
        </div>

        <div className="url-box">
          <label>
            <input 
              type="text" 
              name="url" 
              placeholder="Enter url" 
              value={url} 
              onChange={onChange} />
          </label>
        </div>

        <div className="json-entry-box">
          <label>
            <input 
              type="text" 
              name="body" 
              placeholder="Enter raw JSON" 
              value={body} 
              onChange={onChange} />
          </label>
        </div>

        <button data-testid="button">Submit</button>
      </form>

  </div>
);

Controls.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Controls;