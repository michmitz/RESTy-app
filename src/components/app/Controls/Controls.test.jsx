import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Controls from './Controls';

describe('Controls component', () => {
  afterEach(() => cleanup());

  it('renders Controls', () => {
    const { asFragment } = render(<Controls
      url="testurl"
      method="testmethod"
      body="body"
      onChange={() => {}}
      onSubmit={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});