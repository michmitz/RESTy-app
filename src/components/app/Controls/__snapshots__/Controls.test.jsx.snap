// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders Controls 1`] = `
<DocumentFragment>
  <div
    class="controls-div"
  >
    <form
      class="method-form"
    >
      <div
        class="radio"
      >
        <label>
          <input
            name="method"
            type="radio"
            value="GET"
          />
          GET
        </label>
      </div>
      <div
        class="radio"
      >
        <label>
          <input
            name="method"
            type="radio"
            value="POST"
          />
          POST
        </label>
      </div>
      <div
        class="radio"
      >
        <label>
          <input
            name="method"
            type="radio"
            value="PUT"
          />
          PUT
        </label>
      </div>
      <div
        class="radio"
      >
        <label>
          <input
            name="method"
            type="radio"
            value="DELETE"
          />
          DELETE
        </label>
      </div>
      <div
        class="url-box"
      >
        <label>
          <input
            name="api-url"
            placeholder="Enter url"
            type="text"
            value="testurl"
          />
        </label>
      </div>
      <div
        class="json-entry-box"
      >
        <label>
          <input
            name="body"
            placeholder="Enter raw JSON"
            type="text"
            value="body"
          />
        </label>
      </div>
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
